### PR TITLE
Extend default contribution guidelines

### PR DESCRIPTION
**Pull request by timja**
inspired by 
https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/CONTRIBUTING.md and
https://github.com/jenkinsci/archetypes/pull/155#discussion_r444263393

@jglick @oleg-nenashev @jetersen

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -19,3 +29,66 @@ If you are a newcomer contributor and have any questions, please do not hesitate
...
+
+- For larger contributions create an issue for any required discussion	
+- Implement solution on a branch in your fork	
+- Make sure to include issue number (if created) in commit message, and make the message speak for itself	
+- Once you're done create a pull request and ask at least one of the maintainers for review	
+  - Remember to title your pull request properly as it is used for release notes	
+
+## Run Locally	
+
+Prerequisites: _Java_ and _Maven_
```
Comment (oleg-nenashev) [inline CONTRIBUTING.md:43]: Add a disclaimer about the other flows? We have 50_ repos with Gradle.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
-# Contributing to Jenkins
+# Contributing
 
+**Never report security issues on GitHub or other public channels (Gitter/Twitter/etc.), 	
+follow the instruction from [Jenkins Security](https://jenkins.io/security/) to
```
Comment (oleg-nenashev) [inline CONTRIBUTING.md:4]:
```suggestion
follow the instruction from [Jenkins Security](https://www.jenkins.io/security/#reporting-vulnerabilities) to 	
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
-# Contributing to Jenkins
+# Contributing
 
+**Never report security issues on GitHub or other public channels (Gitter/Twitter/etc.),
```
Comment (oleg-nenashev) [inline CONTRIBUTING.md:3]:
```suggestion
**Never report security issues on GitHub, public Jira issues or other public channels (Gitter/Twitter/etc.), 	
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,5 +1,8 @@
-# Contributing to Jenkins
+# Contributing
```
Comment (oleg-nenashev) [inline CONTRIBUTING.md:2]: Not sure this generic doc should start from reporting issues. An section is fine, but maybe it should start from generic "In the Jenkins project we appreciate any kind of contributions" below,

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -19,3 +29,66 @@ If you are a newcomer contributor and have any questions, please do not hesitate
 
 * [Jenkins: Participate and Contribute](https://jenkins.io/participate/)
 * [Slides: Contributing to Jenkins - it is all about you](https://docs.google.com/presentation/d/1JHgVzWZAx95IsUAZp8OoyCQGGkrCjzUd7eblwd1Y-hA/edit?usp=sharing)
+
+### Source code contribution ways of working	
+
+- For larger contributions create an issue for any required discussion	
+- Implement solution on a branch in your fork	
+- Make sure to include issue number (if created) in commit message, and make the message speak for itself
```
Comment (jglick) [inline CONTRIBUTING.md:37]:
```suggestion
- Make sure to include issue ID (if created) in commit message, and make the message speak for itself	
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -19,3 +29,66 @@ If you are a newcomer contributor and have any questions, please do not hesitate
...
+  ```	
+  - Use the alternate Java 8.	
+
+  ```shell	
+  export JAVA_HOME=`/usr/libexec/java_home -v 1.8`	
+  echo $JAVA_HOME	
+  ```	
+
+  ```text	
+  /Library/Java/JavaVirtualMachines/jdk1.8.0_252.jdk/Contents/Home
```
Comment (jglick) [inline CONTRIBUTING.md:58]:
```suggestion
  ```console	
  $ export JAVA_HOME=`/usr/libexec/java_home -v 1.8`	
  $ echo $JAVA_HOME	
  /Library/Java/JavaVirtualMachines/jdk1.8.0_252.jdk/Contents/Home
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -19,3 +29,66 @@ If you are a newcomer contributor and have any questions, please do not hesitate
...
+  ```	
+
+- Ensure Maven > 3.6.0 is installed and included in the PATH environment variable.	
+
+  ```
```
Comment (jglick) [inline CONTRIBUTING.md:63]:
```suggestion
  ```shell
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -19,3 +29,66 @@ If you are a newcomer contributor and have any questions, please do not hesitate
...
+```shell	
+mvn hpi:run	
+```	
+
+```text	
+...	
+INFO: Jenkins is fully up and running	
+```	
+
+- Open <http://localhost:8080/jenkins/> to test the plugin locally.
```
Comment (jglick) [inline CONTRIBUTING.md:88]: suggest https://github.com/jenkinsci/maven-hpi-plugin/pull/178 instead

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -19,3 +29,66 @@ If you are a newcomer contributor and have any questions, please do not hesitate
...
+### IDE configuration	
+
+See [IDE configuration](https://jenkins.io/doc/developer/development-environment/ide-configuration/)	
+
+### CLI	
+
+- Use the below commands.	
+
+```shell
+mvn install -P quick-build
```
Comment (jglick) [inline CONTRIBUTING.md:76]: You do not need to do this prior to `hpi:run`.

Comment (timja) [reply CONTRIBUTING.md:73]: hmm you will for multi-module projects, I'll move it there

**Pull request merged**